src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double CAPACITY_COEFFICIENT = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ARRAY_START_POSITION_INDEX = 0;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
```

Comment:
better name it checkCapacity, as it checks if grow is needed and  dosen't grow on every call